pkg/frontend/querymiddleware/retry.go

```diff
@@ -32,17 +34,21 @@ func newRetryMiddlewareMetrics(registerer prometheus.Registerer) *retryMiddlewar
 	}
 }
 
+func (m *retryMiddlewareMetrics) Observe(v float64) {
+	m.retriesCount.Observe(v)
+}
```

Comment:
What I meant is that the entire `retryMiddlewareMetrics` struct (and this method) isn't really necessary, as we can just return the histogram we've created. But it's not a big deal.

Reply:
That struct was in the code already.  I imagine it is following some common pattern, or being ready to collect more than one metric?
Either way, changing the way that works seems beyond what I want to achieve here; should be a separate PR.

Reply:
Agree, I'm not blocking on this. 👍